src/Ushahidi/Modules/V5/Common/Authorize.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Ushahidi\Modules\V5\Common;
+
+use App\Auth\GenericUser;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Contracts\Auth\Access\Gate;
+use Ushahidi\Modules\V5\Common\UserContext;
+
+trait Authorize
```

Comment:
i would convert it to a class and use as an actual dependency. This way it's easier to follow and actually modify behaviour later on. And to test it properly.